### PR TITLE
Marketplace: Add upsell nudge at plugin categories and search results

### PR DESCRIPTION
**Pull request by paulopmt1**
#### Proposed Changes

* Added the upgrade nudge at the plugin categories page and the search results page.

#### Testing Instructions

With a simple site without a business plan, test the upsell nudge at:

- /plugins page at `Discover` category
- /plugins page at **any** other category
- /plugins page at the search results page

You should always see this upsell nudge:

image

#### Pre-merge Checklist

- [ ] ~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~
- [ ] ~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~

Related to 14613-gh-Automattic/woocommerce.com

### CODE REVIEW

File: client/my-sites/plugins/search-box-header/style.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .search-box-header {
-	margin: 40px 0 48px;
+	padding: 40px 0 48px;
```
Comment (paulopmt1) [inline client/my-sites/plugins/search-box-header/style.scss:2]: We need to change from margin to padding or order to the search results page to work as expected:


image

**Review by DustyReagan — APPROVED** [review]
Looks good! 👍

**Review by candy02058912 — APPROVED** [review]
LGTM 👍

**Comment by paulopmt1** [comment]
Thanks for the quick review and merging it, folks!

**Pull request merged**